### ISSUE TITLE
data leakage in predict_next_purchases

### ISSUE THREAD

**Issue by datajanko** [Featuretools/predict-next-purchase#5]
Hey,

in the notebook, when using: 

    clf.fit(X, y)
    top_features = utils.feature_importances(clf, features_encoded, n=20)

we introduce data leakage, since we select the features on the whole data set.
With scikit-learn's pipelines it's possible to select the 20 (rather a fraction o features) best features for each fold with [select from model](http://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.SelectFromModel.html#sklearn.feature_selection.SelectFromModel)

With the current set up, you are probably overestimating the AUC.
Besides, cross val score assumes IID samples. However, this will clearly not be the case, since one entity has typically several occurences. I think some thing like [time series split](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html#sklearn.model_selection.TimeSeriesSplit) or rather an adaption (since we don't have time series in a classical way but rather time slices) should be the correct thing to use here.

Comments on those issues?

Currently, at work, I have the same issues, so I really appreciate the library you developed so far. I haven't seen something similar so far. So thumbs up in any case

**Comment by lstmemery** [comment]
I've noticed that a lot of feature engineering tutorials make this mistake (not just featuretools). I think featuretools handles this through cutoff times https://docs.featuretools.com/automated_feature_engineering/handling_time.html

**Comment by datajanko** [comment]
You are right. Actually, the other examples show the correct usage of "backtesting". Shouldn't we at least make a remark here. Changing this, completely, will provide different kaggle results and thus requires more effort to change things.
